src/Admin/DevToolsUserAccess.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Class DevToolsUserAccess.
+ *
+ * @since 1.6.0
+ *
+ * @package AMP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Error;
+use WP_User;
+
+/**
+ * Class DevToolsUserAccess
+ *
+ * @since 1.6.0
+ */
+final class DevToolsUserAccess implements Service, Registerable {
+
+	/**
+	 * User meta key enabling or disabling developer tools.
+	 *
+	 * @var string
+	 */
+	const USER_FIELD_DEVELOPER_TOOLS_ENABLED = 'amp_dev_tools_enabled';
+
+	/**
+	 * Sets up hooks.
+	 */
+	public function register() {
+		add_action( 'rest_api_init', [ $this, 'register_rest_field' ] );
```

Comment:
As this service is only needed for REST requests, it would be preferrable to make it "delayed" to only register at the `rest_api_init` action.

Right now, it always instantiates and the hooks int `rest_api_init` itself.

But for cases like this, the service container has smart behavior that adds an optimization.
If you add the `Delayed` interface with a registration action of `rest_api_init` instead, it will only *instantiate* at the `rest_api_init` action instead. So, as the instantiation itself is moved to that action, the service will *not instantiate* when the current request is not a REST request.

Reply:
Yes, that makes sense. I've updated to add `Delayed`. With this approach, I suppose there's no need to add callbacks to the `rest_api_init` hook, so I also changed `register` to call the `register_rest_field` method directly.

Reply:
Yes, exactly, that's how I would have built it as well.

Reply:
Oh, I assumed now you meant you've used the WordPress `register_rest_field()` method immediately within `DevToolsUserAccess::register()`. There's no need for the `DevToolsUserAccess::register_rest_field()` method anymore, that is just a superfluous method call overhead.

Reply:
I've added it back in because we will need it later, to be able to add the user profile field.